tests/uri_test.c

```diff
@@ -509,7 +509,7 @@ static int s_test_uri_ipv6_parse(struct aws_allocator *allocator, void *ctx) {
         aws_byte_cursor_from_c_str("[2001:db8:85a3:8d3:1319:8a2e:370:7348%25en0]:443");
     ASSERT_BIN_ARRAYS_EQUALS(expected_authority.ptr, expected_authority.len, uri.authority.ptr, uri.authority.len);
 
-    struct aws_byte_cursor expected_host = aws_byte_cursor_from_c_str("[2001:db8:85a3:8d3:1319:8a2e:370:7348%25en0]");
```

Comment:
can you also verify, whether we should be stripping zone idx or not? what do other libs do?

Reply:
It is included 
<img width="1416" alt="image" src="https://github.com/awslabs/aws-c-common/assets/7613045/19bdf4b1-b7a4-4c93-ad16-17f39f311c93">